client/src/components/ModelViewerComponent/ModelViewerComponent.test.js

```diff
@@ -0,0 +1,307 @@
+
+  await findByText(container, "Floor");
+  const nodeList = container.querySelectorAll(".mv_listItem");
+  expect(nodeList.length).toBe(1);
+  expect(retrieveModels).toHaveBeenCalledTimes(1);
+});
+
+test("renders model Information", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.getModelById.mockResolvedValue(modelData);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.viewModel");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  await screen.findByText("modelViewerViewComponent.defaultButton");
+  expect(getModel).toHaveBeenCalledTimes(1);
+});
+
+test("delete model", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.deleteModel.mockResolvedValue(modelData);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.deleteModels");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const confirm = await screen.getByTestId("confirm");
+  act(() => {
+    confirm.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  expect(deleteModel).toHaveBeenCalledTimes(1);
+  await waitFor(() => expect(eventService.publishDeleteModel).toHaveBeenCalledTimes(1));
+});
+
+test("create a twin", async () => {
+  configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+  apiService.queryModels.mockResolvedValue(models);
+  apiService.addTwin.mockResolvedValue(mockSuccesResponse);
+  ModelService.prototype.createPayload.mockResolvedValue(mockSuccesResponse);
+  act(() => {render(<ModelViewerComponent showItemMenu="true" />, container);});
+
+  await findByText(container, "Floor");
+  const button = container.querySelector(".ms-CommandBar-overflowButton");
+  act(() => {
+    button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  const options = await screen.findByLabelText("modelViewerItemCommandBarComponent.farItems.createTwin");
+  act(() => {
+    options.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+  });
+  let input = await screen.findByTestId("twinNameInput");
+  input.value = "TestTwin";
+  Simulate.change(input);
```

Comment:
We might want to write a utility function to handle change input values? I.e. make the three lines here reusable.